### PR TITLE
Cache index repo file

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/28863

File: pkg/catalogv2/content/content.go [hunk 1]
```diff
@@ -76,6 +86,15 @@ func (c *Manager) Index(namespace, name string) (*repo.IndexFile, error) {
 		return nil, err
 	}
 
+	c.lock.RLock()
+	if cache, ok := c.IndexCache[fmt.Sprintf("%s/%s", r.status.IndexConfigMapNamespace, r.status.IndexConfigMapName)]; ok {
+		if bytes.Compare(cache.binaryData, cm.BinaryData["content"]) == 0 {
```
Comment (ibuildthecloud) [inline pkg/catalogv2/content/content.go:91]: Why don't you just keep the revision number of the config map. Faster comparison.

File: pkg/catalogv2/content/content.go [hunk 2]
```diff
@@ -95,11 +114,18 @@ func (c *Manager) Index(namespace, name string) (*repo.IndexFile, error) {
...
+		index:      *index,
+		binaryData: cm.BinaryData["content"],
+	}
+	c.lock.Unlock()
+
 	return c.filterReleases(index), nil
 }
 
 func (c *Manager) filterReleases(index *repo.IndexFile) *repo.IndexFile {
-	index.SortEntries()
```
Comment (ibuildthecloud) [inline pkg/catalogv2/content/content.go:102]: Was this redundant? It's really important the data is sorted.

File: pkg/catalogv2/content/content.go (hunk 1 above)
Comment (StrongMonkey) [reply pkg/catalogv2/content/content.go:91]: oh you are right

File: pkg/catalogv2/content/content.go (hunk 2 above)
Comment (StrongMonkey) [reply pkg/catalogv2/content/content.go:102]: https://github.com/rancher/rancher/blob/417bdfd9874f8903c5d9923f78859fb1e313a098/pkg/controllers/dashboard/helm/repo.go#L180 you did sorting at here

**Pull request merged**